test/integration/ComputeFlow.test.ts

```diff
@@ -472,348 +476,397 @@ describe('Compute flow tests', async () => {
+  })
+
+  it('should restart a computeJob on paid environment, by paying only escrow lock for max job duration, because orders for assets are valid and providerFees are still valid', async () => {
+    // we choose the paid env
+    computeEnvs = await ProviderInstance.getComputeEnvironments(providerUrl)
+    const computeEnv = computeEnvs[0]
+    assert(computeEnv, 'Cannot find the paid compute env')
+
+    const resources: ComputeResourceRequest[] = [
+      {
+        id: 'cpu',
+        amount: 2
+      },
+      {
+        id: 'ram',
+        amount: 1000000000
+      },
+      {
+        id: 'disk',
+        amount: 0
+      }
+    ]
+
+    const assets: ComputeAsset[] = [
+      {
+        documentId: resolvedDdoWith2mTimeout.id,
+        serviceId: resolvedDdoWith2mTimeout.services[0].id,
+        transferTxId: paidEnvDatasetTxId
+      }
+    ]
+    const algo: ComputeAlgorithm = {
+      documentId: resolvedAlgoDdoWith2mTimeout.id,
+      serviceId: resolvedAlgoDdoWith2mTimeout.services[0].id,
+      transferTxId: paidEnvAlgoTxId
+    }
+
+    providerInitializeComputeResults = await ProviderInstance.initializeCompute(
+      assets,
+      algo,
+      computeEnv.id,
+      paymentToken,
+      computeJobDuration,
+      providerUrl,
+      consumerAccount,
+      resources
+    )
+    assert(
+      providerInitializeComputeResults.algorithm.validOrder,
+      'We should have a valid order for algorithm'
+    )
+    assert(
+      !providerInitializeComputeResults.algorithm.providerFee,
+      'We should not pay providerFees again for algorithm'
+    )
+    assert(
+      providerInitializeComputeResults.datasets[0].validOrder,
+      'We should have a valid order for dataset'
+    )
+    assert(
+      !providerInitializeComputeResults.datasets[0].providerFee,
+      'We should not pay providerFees again for dataset'
+    )
+    algo.transferTxId = providerInitializeComputeResults.algorithm.validOrder
+    assets[0].transferTxId = providerInitializeComputeResults.datasets[0].validOrder
+    assert(
+      algo.transferTxId === paidEnvAlgoTxId &&
+        assets[0].transferTxId === paidEnvDatasetTxId,
+      'We should use the same orders, because no fess must be paid'
+    )
+
+    const computeJobs = await ProviderInstance.computeStart(
+      providerUrl,
+      consumerAccount,
+      computeEnv.id,
+      assets,
+      algo,
+      computeJobDuration,
+      paymentToken,
+      resources
+    )
+    assert(computeJobs, 'Cannot start compute job')
+  })
+
+  // move to reuse Orders
+
+  delay(180000)
+
+  it('should start a computeJob using the paid environment, by paying the assets providerFees (reuseOrder) and paying escrow lock for max job duration', async () => {
+    // we choose the paid env
+    computeEnvs = await ProviderInstance.getComputeEnvironments(providerUrl)
+    const computeEnv = computeEnvs[0]
+    assert(computeEnv, 'Cannot find the paid compute env')
+
+    const resources: ComputeResourceRequest[] = [
```

Comment:
i think it would be nice to have a test where the users triyes to use a higher value of the resources available and the compute job should fail, and we cover this in a test

Reply:
I tested that in previous commits, in ocean-node logs it appears as expected https://github.com/oceanprotocol/ocean.js/actions/runs/14848030257/job/41686162694#step:21:10, but ocean.js could not capture the node's response for initialize (because I tried to return it)

Reply:
re-added